misc/gen-compile-commands.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+
+import os
+import subprocess as sp
+import sys
+
+if not os.path.exists('Makefile'):
+    print('error: Makefile not found.')
+    sys.exit(-1)
+
+def do_dry_make():
+    make_cmds = 'make V=1 --dry-run'
+    with sp.Popen(make_cmds.split(), stdout=sp.PIPE) as p:
+        lines = p.communicate()[0].decode(errors='ignore').split('\n')
+        return lines
+
+with open('compile_commands.json', 'w') as f:
+    f.write('[')
+    needs_comma = False
+    directory = os.getcwd()
+    lines = do_dry_make()
+    for line in lines:
+        command = line
+        if command[:3] != 'gcc':
```

Comment:
Why do you need this?  I guess you wanted clang as well, no?

Reply:
I've added a comment as follows.

> We're only interested in compilation commands so skip otherwise.